nima_core/federation/pg_graph.py

```
@@ -0,0 +1,485 @@
+    }
+    # Gate on reachability: only include neighbors whose edges live in
+    # entities mentioned by at least one memory the caller owns.
+    query = f"""
+        SELECT * FROM cypher('{NIMA_HUB_GRAPH_NAME}', $$
+            MATCH (m:Memory)-[:MENTIONS]->(src:Entity {{name: $entity_name}})
+            WHERE m.namespace IN $namespaces
+            MATCH (src)-[r]-(neighbor:Entity)
+            RETURN DISTINCT neighbor.name, neighbor.entity_type, type(r)
+            LIMIT $limit
+        $$, %s) AS (name agtype, entity_type agtype, rel agtype)
+    """
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)